src/manifests_workflow/input_manifests.py

```diff
@@ -84,92 +85,75 @@ def update(
 
             # generate new manifests
-            for release_version in sorted(main_versions.keys() - known_versions):
-                self.write_manifest(release_version, main_versions[release_version])
-                self.add_to_cron(release_version)
-                self.add_to_versionincrement_workflow(release_version)
-
-    def create_manifest(self, version: str, components: List = []) -> InputManifest:
-        templates_base_path = os.path.join(self.manifests_path(), "templates")
-        template_version_folder = version.split(".")[0] + ".x"
-        template_full_path = os.path.join(templates_base_path, self.prefix, template_version_folder, "manifest.yml")
-        if not os.path.exists(template_full_path):
-            template_full_path = os.path.join(templates_base_path, self.prefix, "default", "manifest.yml")
+            new_versions = sorted(branch_versions.keys() - known_versions, key=version_parse)
+            logging.info(f"New Versions: {new_versions}")
+            for new_version_entry in new_versions:
+                self.write_manifest(new_version_entry, branch_versions[new_version_entry], known_versions)
+                self.add_to_cron(new_version_entry)
+                self.add_to_versionincrement_workflow(new_version_entry)
+                known_versions.append(new_version_entry)
+
+    def create_manifest(self, version: str, branch: str, known_versions: List[str]) -> InputManifest:
+        # If  : No known_versions manifests exist or new version smaller than the min(known_versions), create new manifests from the templates
+        #       (1.0.0-3.0.0 based on template 1.x-3.x, 4.0.0+ from default.x, previous behavior)
+        # Else: Create new manifests based on the latest version before the new version
+        #       (2.12.1 from 2.12.0, 2.13.0 from 2.12.1, 3.0.0 from 2.13.0, 4.0.0 from 3.0.0, etc.)
+        if not known_versions or Version(version) < Version(min(known_versions)):
+            logging.info("No previous version exist before {version}, create with templates")
+            templates_base_path = os.path.join(self.manifests_path(), "templates")
+            template_version_folder = version.split(".")[0] + ".x"
+            template_full_path = os.path.join(templates_base_path, self.prefix, template_version_folder, "manifest.yml")
+            if not os.path.exists(template_full_path):
+                template_full_path = os.path.join(templates_base_path, self.prefix, "default", "manifest.yml")
+        else:
+            previous_versions = [v for v in known_versions if Version(v) < Version(version)]
+            base_version = max(previous_versions, key=version_parse)
+            logging.info(f"Base Version: {base_version} is the highest version before {version}")
+            template_full_path = os.path.join(self.manifests_path(), base_version, f"{self.prefix}-{base_version}.yml")
+            if not os.path.exists(template_full_path):
```

Comment:
Should we error it out here and throw an exception?
Are we expecting to find manifests for newer versions in the legacy path.
The logic here assumes that there is a possibility of us doing patch release of older versions apart from current GA version?

Reply:
Exactly. As I mentioned in previous comment, if someone wants to release 2.2.3, but 2.2.2 is already in legacy folder, the code can still find 2.2.2 in legacy folder to create 2.2.3.

Reply:
Especially the code to retrieve the `known_versions` is already checking both manifests, and legacy-manifests folders.